OSDOCS-11794: ROSA CLI security refinements adding CLI output examples (#81844)
openshift/openshift-docs: modules/rosa-sts-account-wide-roles-and-policies.adoc

----
====

[discrete]
[id="rosa-sts-account-wide-roles-and-policies-example-cli-output-for-policies-attached-to-a-role_{context}"]
==== Example CLI output for policies attached to a role 

> When a policy is attached to a role, the ROSA CLI displays confirmation output. The output depends on the type of policy.

Reviewer: ```suggestion
When a policy is attached to a role, the ROSA CLI displays a confirmation output. The output depends on the type of policy.
```

> * If a policy is a trust policy, it outputs the content of the policy.

Reviewer: ```suggestion
* If the policy is a trust policy, the ROSA CLI outputs the content of the policy.
```

Reviewer 2: should be:
* If the policy is a trust policy, the ROSA CLI outputs the role name and the content of the policy.

> ** For the target role with policy attached, ROSA CLI outputs the role name and the console URL of it. 

Reviewer: ```suggestion
** For the target role with a policy attached, the ROSA CLI outputs the role name and console URL of the target role. 
```

.Target role with policy attached example output
[source,terminal]
----
I: Attached trust policy to role 'testrole-Worker-Role(https://console.aws.amazon.com/iam/home?#/roles/testrole-Worker-Role)': ******************
----

> If the attached policy is a trust policy, ROSA CLI outputs the content of this policy.

Reviewer 3: it's better to mention the policy type explicitly like below:

if a policy is a **trust policy**, then: output content
if a policy is a **permission policy**, then:
     if this policy is aws managed policy, then: output name+public link
     if this policy is not a managed policy, then: output arn

> ** If the attached policy is a trust policy, ROSA CLI outputs the content of this policy.

Reviewer: ```suggestion
** If the attached policy is a trust policy, the ROSA CLI outputs the content of this policy.
```

.Trust policy example output
[source,terminal]
----
I: Attached trust policy to role 'test-Support-Role': {"Version": "2012-10-17", "Statement": [{"Action": ["sts:AssumeRole"], "Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::000000000000:role/RH-Technical-Support-00000000"]}}]} 
----

> If the attached policy is an AWS managed policy, ROSA CLI outputs the public link of this policy. 

Reviewer 3: ROSA CLI outputs the **name and** public link of this policy.

> * If a policy is a permission policy, it outputs the name and public link of this policy or the ARN depending on whether or not the policy is an AWS managed policy or non-managed.

Reviewer: ```suggestion
* If the policy is a permission policy, the ROSA CLI outputs the name and public link of this policy or the ARN depending on whether or not the policy is an AWS managed or unmanaged policy.
```

Reviewer: ARN, AWS, ROSA CLI -> expand first occurence ??

Reviewer 2: rather than non-managed or unmanaged policy, the appropriate term in AWS is
"customer-managed policy"

> ** If the attached policy is an AWS managed policy, ROSA CLI outputs the name and public link of this policy. 

Reviewer: ```suggestion
** If the attached policy is an AWS managed policy, the ROSA CLI outputs the name and public link of this policy. 
```

Reviewer 2: should be:
** If the attached policy is an AWS managed policy, the ROSA CLI outputs the name and public link of this policy and the role it's attached to.

+
.AWS managed policy example output
[source,terminal]
----
I: Attached policy 'ROSASRESupportPolicy(https://docs.aws.amazon.com/aws-managed-policy/latest/reference/ROSASRESupportPolicy)' to role 'test-HCP-ROSA-Support-Role(https://console.aws.amazon.com/iam/home?#/roles/test-HCP-ROSA-Support-Role)' 
----

> ** If the attached policy is non-managed policy, ROSA CLI outputs the ARN of this policy. 

Reviewer: ```suggestion
** If the attached policy is an unmanaged policy, the ROSA CLI outputs the ARN of this policy. 
```

Reviewer 2: same comment as in https://github.com/openshift/openshift-docs/pull/81844/files#r1764645820

Reviewer 2: should be:
** If the attached policy is an AWS managed policy, the ROSA CLI outputs the name and public link of this policy and the role it's attached to.

+

> .Non-managed policy example output

Reviewer 2: same comment as in https://github.com/openshift/openshift-docs/pull/81844/files#r1764645820
